lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -268,137 +268,224 @@ private static Domain getDomain(
-            List<Range> ranges = new ArrayList<>(minimums.size());
-            for (int i = 0; i < minimums.size(); i++) {
-                long min = asLong(minimums.get(i));
-                long max = asLong(maximums.get(i));
-                if (isStatisticsOverflow(type, min, max)) {
-                    return Domain.create(ValueSet.all(type), hasNullValue);
+            if (valuesList instanceof StatisticsRangeList statisticsRanges) {
+                List<Range> ranges = new ArrayList<>(valuesList.size());
+                for (StatisticsRange statisticsRange : statisticsRanges.rangesList()) {
+                    long min = asLong(statisticsRange.min());
+                    long max = asLong(statisticsRange.max());
+                    if (isStatisticsOverflow(type, min, max)) {
+                        return Domain.create(ValueSet.all(type), hasNullValue);
+                    }
+                    ranges.add(Range.range(type, min, true, max, true));
                 }
-
-                ranges.add(Range.range(type, min, true, max, true));
+                return Domain.create(ValueSet.ofRanges(ranges), hasNullValue);
             }
 
-            return Domain.create(ValueSet.ofRanges(ranges), hasNullValue);
+            DiscreteValuesList discreteValuesList = (DiscreteValuesList) valuesList;
+            List<Long> values = new ArrayList<>(valuesList.size());
+            for (Object value : discreteValuesList.values()) {
+                long valueAsLong = asLong(value);
+                if (isStatisticsOverflow(type, valueAsLong, valueAsLong)) {
+                    return Domain.create(ValueSet.all(type), hasNullValue);
+                }
+                values.add(valueAsLong);
+            }
+            return Domain.create(ValueSet.copyOf(type, values), hasNullValue);
         }
 
-        if (type instanceof DecimalType) {
-            DecimalType decimalType = (DecimalType) type;
-            List<Range> ranges = new ArrayList<>(minimums.size());
+        if (type instanceof DecimalType decimalType) {
             if (decimalType.isShort()) {
-                for (int i = 0; i < minimums.size(); i++) {
-                    Object min = minimums.get(i);
-                    Object max = maximums.get(i);
+                if (valuesList instanceof StatisticsRangeList statisticsRanges) {
+                    List<Range> ranges = new ArrayList<>(valuesList.size());
+                    for (StatisticsRange statisticsRange : statisticsRanges.rangesList()) {
+                        Object min = statisticsRange.min();
+                        Object max = statisticsRange.max();
 
-                    long minValue = min instanceof Binary ? getShortDecimalValue(((Binary) min).getBytes()) : asLong(min);
-                    long maxValue = min instanceof Binary ? getShortDecimalValue(((Binary) max).getBytes()) : asLong(max);
```

Comment:
there was a bug here, right?

can you please fix it as a prep commit?

Reply:
It's not possible for the min and max values to be of different types in the file, it's more of a consistency/readability fix